test/common.js

```diff
@@ -0,0 +1,19 @@
+var yaml = require('js-yaml');
+
+module.exports.getDiag = function (body) {
+    var yamlStart = body.indexOf('  ---');
+    var yamlEnd = body.indexOf('  ...\n');
+    var diag = body.slice(yamlStart, yamlEnd).split('\n').map(function (line) {
+        return line.slice(2);
+    }).join('\n');
+
+    // The stack trace will vary depending on where the code is run, so just
+    // strip it out.
+    var withStack = yaml.safeLoad(diag);
+    delete withStack.stack;
+    return withStack;
+}
+
+module.exports.stripFullStack = function (output) {
+    return output.replace(/^\W+at.*:\d+:\d+.*$\W/gm, '');
```

Comment:
So, here's the first-order problems that checking full stacks with every test would create:

1. The file names will change depending on the absolute directory of the git clone.
2. Any changes to line position (adding or removing things from a test or even potentially to a file that contains code as part of a test traceback) will cause tests to fail.

These are the two biggest problems, and could be easily fixed by stripping the relevant volatile part of the stack. The deeper issue that can't be resolved is the structure of the stack itself. Any refactoring that adds or removes a method to any "failing stack" will now fail the test.

As a result, many trivial refactors would then result in lots of pointless new test failures. It was annoying enough to go through all these tests myself once to fix this issue. I don't want to put that burden on every future author.

Does that make sense? Maybe I'm just not understanding what you're expecting to happen when normalizing output across engines. Is there some deeper property of the stacks we want to test everywhere?